server/SecondSurvey.cs

```diff
@@ -393,10 +393,9 @@ public static int Importance(World world)
                 i += 1;
             if (world.PopulationExponent <= 6)
                 i -= 1;
-            if ((world.Bases.Contains('N') || world.Bases.Contains('K')) &&
-                (world.Bases.Contains('S') || world.Bases.Contains('V')))
-                i += 1;
-            if (world.Bases.Contains('W'))
+            if ((world.Bases.Contains('D') || world.Bases.Contains('W') || world.Bases.Contains('X')) ||
+                ((world.Bases.Contains('K') || world.Bases.Contains('N') || world.Bases.Contains('T')) &&
+                (world.Bases.Contains('C') || world.Bases.Contains('R') || world.Bases.Contains('S') || world.Bases.Contains('V'))))
```

Comment:
If the style is too different for normal use then feel free to reject, but I would think something like this would make the intent more clear:
```
private const string NAVAL_BASES = "KNT";
private const string SCOUT_BASES = "CRSV";
private const string ADVANCED_BASES = "DWX";
...
if (ADVANCED_BASES.Any(a => world.Bases.Contains(a))
    || (NAVAL_BASES.Any(n => world.Bases.Contains(n)) && SCOUT_BASES.Any(s => world.Bases.Contains(s))))
    i += 1;
```